File: core/src/main/java/in/testpress/util/Assert.java
package in.testpress.util;

import android.content.Context;
import android.support.annotation.NonNull;

public class Assert {

    public static void assertNotNull(@NonNull String message, Object object) {
        if (object == null) {
            throw new IllegalArgumentException(message);
        }
    }

    public static void assertNotNullAndNotEmpty(@NonNull String message, String string) {
        if (string == null || string.isEmpty()) {
            throw new IllegalArgumentException(message);
        }
    }

    public static void assertContextNotNull(Context context) {
        if (context == null) {
            throw new IllegalArgumentException("Context must not be null.");
        }
    }
}

File: core/src/main/java/in/testpress/util/Assert.kt
package `in`.testpress.util

class Assert {

    companion object {

        @JvmStatic
        fun assertNotNull(message: String, any: Any?) {
            require(any != null) { message }
        }

        @JvmStatic
        fun assertNotNullAndNotEmpty(message: String, string: String?) {
            require(!string.isNullOrEmpty()) { message }
        }
    }
}